draft-ietf-acme-acme.md

```diff
@@ -1751,7 +1752,10 @@ reason (optional, int):
 to be used when generating OCSP responses and CRLs. If this field is not set
 the server SHOULD use the unspecified (0) reasonCode value when generating OCSP
 responses and CRLs. The server MAY disallow a subset of reasonCodes from being
-used by the user.
+used by the user. If a request contains a disallowed reasonCode the server MUST
+reject it with the error type "urn:ietf:params:acme:error:badRevocationReason".
+The problem document returned with the error MUST include a "supportedReasons"
```

Comment:
I'm reluctant to add extra infrastructure (the "supportedReasons" field) for a use case that will be exceedingly rare. We don't really expect most clients to implement some sort of fancy menu like "Try one of these other revocation reasons!" Instead how about we just say "the problem detail SHOULD indicate which revocation reasons are allowed."